docker/Dockerfile

```diff
@@ -0,0 +1,111 @@
+RUN tar -xzf gcloud-sdk.tar.gz
+RUN ./google-cloud-sdk/install.sh --quiet
+RUN rm gcloud-sdk.tar.gz
+
+WORKDIR /app/oppia
+
+# installing python dependencies from the requirements_dev.txt file
+COPY requirements_dev.in .
+COPY requirements_dev.txt .
+
+RUN pip-compile --generate-hashes requirements_dev.in
+RUN pip install --require-hashes --no-deps -r requirements_dev.txt
+
+## installing packages from the package.json file
+COPY package.json .
+COPY scripts/linters/custom_eslint_checks ./scripts/linters/custom_eslint_checks
+
+RUN npm install -g yarn
+# mathjs ESOCKETTIMEDOUT error fix -- https://stackoverflow.com/questions/55845756/esockettimedout-error-while-yarn-install-command
+RUN yarn install --network-timeout 1000000
+
+# installing buf and proto for Linux -- this docker container is based on Linux
+COPY buf.gen.yaml .
+COPY /extensions/classifiers/proto/ ./extensions/classifiers/proto/
+
+# installing third party dependencies
+COPY scripts ./scripts
+COPY /core ./core
+COPY /assets ./assets
+COPY dependencies.json .
+
+# installation for proto buf requires the third party dependencies to be installed (oppia-ml-proto)
+RUN python -m scripts.install_third_party
+
+ENV BUF_LINUX_FILES="buf-Linux-x86_64 protoc-gen-buf-check-lint-Linux-x86_64 protoc-gen-buf-check-breaking-Linux-x86_64" \
+    PROTOC_LINUX_FILE='protoc-3.13.0-linux-x86_64.zip' \
+    BUF_DIR='/app/buf-0.29.0' \
+    PROTOC_DIR='/app/buf-0.29.0/protoc' \
+    BUF_BASE_URL='https://github.com/bufbuild/buf/releases/download/v0.29.0/' \
+    PROTOC_URL='https://github.com/protocolbuffers/protobuf/releases/download/v3.13.0'
+
+# Download BUF_LINUX_FILES and PROTOC_LINUX_FILE
+# set the shell as /bin/bash explicitly. This is required to make use of the array syntax correctly
+SHELL ["/bin/bash", "-c"]
+RUN IFS=' ' read -ra BUF_FILES <<< "$BUF_LINUX_FILES"; \
+    for bin_file in "${BUF_FILES[@]}"; do \
+        wget -P $BUF_DIR $BUF_BASE_URL/$bin_file; \
+    done
+
+RUN wget -P $BUF_DIR $PROTOC_URL/$PROTOC_LINUX_FILE
+RUN unzip $BUF_DIR/$PROTOC_LINUX_FILE -d $PROTOC_DIR \
+    && rm $BUF_DIR/$PROTOC_LINUX_FILE
+
+RUN chmod -R 744 $BUF_DIR \
+    && chmod -R 744 $PROTOC_DIR
+
+# compiling the protobuf files
+ENV PROTOC_FILES_PATH='/app/oppia/third_party/oppia-ml-proto-0.0.0'
+
+RUN mkdir /app/oppia/proto_files
+RUN /app/buf-0.29.0/protoc/bin/protoc \
+    --plugin=protoc-gen-ts=/app/oppia/node_modules/.bin/protoc-gen-ts \
+    --ts_out=/app/oppia/extensions/classifiers/proto/ \
+    --js_out=import_style=commonjs,binary:/app/oppia/extensions/classifiers/proto/ \
+    --python_out=/app/oppia/proto_files \
+    --proto_path=$PROTOC_FILES_PATH \
+    $PROTOC_FILES_PATH/*.proto
+
+RUN sed -i 's/import text_classifier_pb2 as text__classifier__pb2/from . import text_classifier_pb2 as text__classifier__pb2/' /app/oppia/proto_files/training_job_response_payload_pb2.py
```

Comment:
no, this need to be downloaded at this place only as it requires npm packages and python packages too from the requirements_dev.txt, I placed this here after properly testing and then placing where it can be placed in a most optimal way